In [1]:
import json
import pandas as pd
import numpy as np

In [2]:
with open("disease_signatures-v1.0.json",'r') as load_f:
    load_list = json.load(load_f)

In [3]:
load_dict = pd.DataFrame(columns=load_list[0].keys())

for i in range(len(load_list)):
    load_dict = load_dict.append(load_list[i],ignore_index=True)

print(len(load_dict))
load_dict.head()

828


do_id                             cell_type  \
0  DOID:11723  Muscle - Striated (Skeletal) (MMHCC)   
1        None                  Ganglioneuroblastoma   
2        None                                  Nose   
3  DOID:12930                     Myocardial tissue   
4    DOID:104     Peripheral blood mononuclear cell   

                                            pert_ids  umls_cui    geo_id  \
0      [GSM4377, GSM4378, GSM4379, GSM4380, GSM4381]  C0013264    GSE466   
1  [GSM282584, GSM282585, GSM282586, GSM282590, G...  C0028043  GSE11208   
2  [GSM286646, GSM286647, GSM286648, GSM286652, G...  C0276447  GSE11348   
3  [GSM82408, GSM82409, GSM82410, GSM82411, GSM82...  C0007193   GSE3586   
4  [GSM66617, GSM66618, GSM66619, GSM66620, GSM66...  C0004623   GSE3026   

  platform version                                           ctrl_ids  \
0    GPL81     1.0      [GSM4372, GSM4373, GSM4374, GSM4375, GSM4376]   
1   GPL570     1.0  [GSM282582, GSM282583, GSM282587, GSM282588, G...   
2   GPL570     1.0  [GSM286649, GSM286650, GSM286651, GSM286655, G...   
3  GPL3050     1.0  [GSM82393, GSM82394, GSM82395, GSM82396, GSM82...   
4  GPL8300     1.0  [GSM66671, GSM66672, GSM66673, GSM66674, GSM66...   

                                          down_genes  \
0  [[Rpl41, -0.1526295691728592], [Tnnt3, -0.1450...   
1  [[BEX1, -0.08722281455993652], [ZNF423, -0.077...   
2  [[HLA-DRB4, -0.2605530321598053], [BPIFB1, -0....   
3  [[CKM, -0.20369680225849152], [SMPX, -0.177489...   
4  [[RPS27, -0.20674751698970795], [CD74, -0.1724...   

                                            up_genes  \
0  [[Actc1, 0.1277850866317749], [Lyz1, 0.1037828...   
1  [[RPS18, 0.19893527030944824], [COX2, 0.193066...   
2  [[FCGBP, 0.1933731883764267], [S100A4, 0.19266...   
3  [[MB, 0.3557915985584259], [MYL2, 0.2121689766...   
4  [[HLA-DRB5, 0.26361608505249023], [IGHA2, 0.16...   

                   disease_name organism      id      curator  
0   Duchenne muscular dystrophy    mouse  dz:328  Joel.Dudley  
1            Nicotine addiction    human  dz:325  Joel.Dudley  
2          Rhinovirus infection    human  dz:324  Joel.Dudley  
3        dilated cardiomyopathy    human  dz:323  Joel.Dudley  
4  bacterial infectious disease    human  dz:322  Joel.Dudley

In [4]:
#load_dict.to_csv('disease_signatures-v1.0.csv')

### Use only human-organized data

In [4]:
load_dict = load_dict[load_dict['organism']=="human"]

In [5]:
load_dict.tail()

do_id                                          cell_type  \
819  DOID:3969  Papillary thyroid carcinoma tumors (with a BRA...   
820  DOID:3969  Papillary thyroid carcinoma tumors (without a ...   
823  DOID:3151          Skin from SCCs and Normal human epidermis   
824   DOID:684  Hepatocellular carcinoma and Adjacent non-tumo...   
826       None  Anaplastic thyroid carcinoma tissue and Normal...   

                                              pert_ids  umls_cui    geo_id  \
819  [GSM1327242, GSM1327243, GSM1327244, GSM132724...  C0238463  GSE54958   
820  [GSM1327231, GSM1327232, GSM1327233, GSM132723...  C0238463  GSE54958   
823  [GSM1098449, GSM1098450, GSM1098451, GSM109845...  C0553723  GSE45164   
824  [GSM1481293, GSM1481295, GSM1481297, GSM148129...  C0279607  GSE60502   
826  [GSM1588026, GSM1588027, GSM1588028, GSM158802...  C0238461  GSE65144   

    platform version                                           ctrl_ids  \
819  GPL6244     1.0  [GSM1327256, GSM1327257, GSM1327258, GSM132725...   
820  GPL6244     1.0  [GSM1327256, GSM1327257, GSM1327258, GSM132725...   
823   GPL571     1.0               [GSM1098446, GSM1098447, GSM1098448]   
824    GPL96     1.0  [GSM1481292, GSM1481294, GSM1481296, GSM148129...   
826   GPL570     1.0  [GSM1588038, GSM1588039, GSM1588040, GSM158804...   

                                            down_genes  \
819  [[MT1F, -0.1966540813446045], [REXO1L1P, -0.12...   
820  [[REXO1L1P, -0.10428610444068909], [HBA2, -0.1...   
823  [[CST6, -0.16002681851387024], [KRT2, -0.13491...   
824  [[MT2A, -0.17223264276981354], [CYP2B7P, -0.16...   
826  [[TG, -0.4695512056350708], [TFF3, -0.12417372...   

                                              up_genes  \
819  [[FN1, 0.22676201164722443], [SLC34A2, 0.13122...   
820  [[TG, 0.11662885546684265], [CTSB, 0.103684581...   
823  [[DEFB4A, 0.2018461525440216], [S100A7, 0.1836...   
824  [[SPINK1, 0.16987518966197968], [GPC3, 0.10506...   
826  [[LGALS1, 0.20619834959506989], [TMSB10, 0.178...   

                     disease_name organism      id curator  
819   papillary thyroid carcinoma    human  dz:653  cadimo  
820   papillary thyroid carcinoma    human  dz:652  cadimo  
823  skin squamous cell carcinoma    human  dz:657  cadimo  
824      hepatocellular carcinoma    human  dz:656  cadimo  
826  anaplastic thyroid carcinoma    human  dz:654  cadimo

In [6]:
print(len(load_dict))

555


### Extracting up- and down-regulated gene set data

In [34]:
dis_DEG = pd.DataFrame()
for i in range(len(load_dict)):
    tem_dic = {}
    
    down_genes = load_dict.iloc[i,8] #down_genes
    down_len = len(down_genes)
    up_genes = load_dict.iloc[i,9]  #up_genes
    up_len = len(up_genes)
    
    for j in range(down_len):
        tem_dic[down_genes[j][0]]=down_genes[j][1]
    for k in range(up_len):
        tem_dic[up_genes[k][0]]=up_genes[k][1]
    if not pd.isnull(load_dict.iloc[i,3]):         
        genes_df = pd.DataFrame(tem_dic,index=[load_dict.iloc[i,3]])  #disease_UMLS-CUI,include Nan
        #genes_df = pd.DataFrame(tem_dic,index=[load_dict.iloc[i,10]])  #disease_name
        dis_DEG = dis_DEG.append(genes_df)

D:\Anaconda\lib\site-packages\pandas\core\frame.py:7123: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
of pandas will change to not sort by default.

To accept the future behavior, pass 'sort=False'.

To retain the current behavior and silence the warning, pass 'sort=True'.

  sort=sort,


In [35]:
""" signature counts using disease name as index：555
    signature counts using disease CUI id as index：509"""
print(len(dis_DEG))
dis_DEG.head()

509


1110008P14Rik  1190002N15Rik  2'-PDE  2310036O22Rik  2900097C17Rik  \
C0028043            NaN            NaN     NaN            NaN            NaN   
C0276447            NaN            NaN     NaN            NaN            NaN   
C0007193            NaN            NaN     NaN            NaN            NaN   
C0004623            NaN            NaN     NaN            NaN            NaN   
C0085786            NaN            NaN     NaN            NaN            NaN   

          7A5  A1BG  A1CF  A2BP1  A2M  ...  Zmiz2  Zranb2  Zwint  dJ222E13.2  \
C0028043  NaN   NaN   NaN    NaN  NaN  ...    NaN     NaN    NaN         NaN   
C0276447  NaN   NaN   NaN    NaN  NaN  ...    NaN     NaN    NaN         NaN   
C0007193  NaN   NaN   NaN    NaN  NaN  ...    NaN     NaN    NaN         NaN   
C0004623  NaN   NaN   NaN    NaN  NaN  ...    NaN     NaN    NaN         NaN   
C0085786  NaN   NaN   NaN    NaN  NaN  ...    NaN     NaN    NaN         NaN   

          hCG_15200  hCG_1806964  hCG_18290  tcag7.1196  tcag7.1228  \
C0028043        NaN          NaN        NaN         NaN         NaN   
C0276447        NaN          NaN        NaN         NaN         NaN   
C0007193        NaN          NaN        NaN         NaN         NaN   
C0004623        NaN          NaN        NaN         NaN         NaN   
C0085786        NaN          NaN        NaN         NaN         NaN   

          tcag7.1260  
C0028043         NaN  
C0276447         NaN  
C0007193         NaN  
C0004623         NaN  
C0085786         NaN  

[5 rows x 19563 columns]

In [21]:
#dis_DEG.to_csv('dis_hDEG_name.csv',index=True)
#dis_DEG.to_csv('dis_hDEG_id.csv',index=True)

In [2]:
dis_DEG = pd.read_csv('dis_hDEG_name.csv',index_col=0)
dis_DEG.head()

1110008P14Rik  1190002N15Rik  2'-PDE  \
nicotine addiction                           NaN            NaN     NaN   
rhinovirus infection                         NaN            NaN     NaN   
dilated cardiomyopathy                       NaN            NaN     NaN   
bacterial infectious disease                 NaN            NaN     NaN   
idiopathic interstitial pneumonia            NaN            NaN     NaN   

                                   2310036O22Rik  2900097C17Rik  7A5  A1BG  \
nicotine addiction                           NaN            NaN  NaN   NaN   
rhinovirus infection                         NaN            NaN  NaN   NaN   
dilated cardiomyopathy                       NaN            NaN  NaN   NaN   
bacterial infectious disease                 NaN            NaN  NaN   NaN   
idiopathic interstitial pneumonia            NaN            NaN  NaN   NaN   

                                   A1CF  A2BP1  A2M  ...  Zmiz2  Zranb2  \
nicotine addiction                  NaN    NaN  NaN  ...    NaN     NaN   
rhinovirus infection                NaN    NaN  NaN  ...    NaN     NaN   
dilated cardiomyopathy              NaN    NaN  NaN  ...    NaN     NaN   
bacterial infectious disease        NaN    NaN  NaN  ...    NaN     NaN   
idiopathic interstitial pneumonia   NaN    NaN  NaN  ...    NaN     NaN   

                                   Zwint  dJ222E13.2  hCG_15200  hCG_1806964  \
nicotine addiction                   NaN         NaN        NaN          NaN   
rhinovirus infection                 NaN         NaN        NaN          NaN   
dilated cardiomyopathy               NaN         NaN        NaN          NaN   
bacterial infectious disease         NaN         NaN        NaN          NaN   
idiopathic interstitial pneumonia    NaN         NaN        NaN          NaN   

                                   hCG_18290  tcag7.1196  tcag7.1228  \
nicotine addiction                       NaN         NaN         NaN   
rhinovirus infection                     NaN         NaN         NaN   
dilated cardiomyopathy                   NaN         NaN         NaN   
bacterial infectious disease             NaN         NaN         NaN   
idiopathic interstitial pneumonia        NaN         NaN         NaN   

                                   tcag7.1260  
nicotine addiction                        NaN  
rhinovirus infection                      NaN  
dilated cardiomyopathy                    NaN  
bacterial infectious disease              NaN  
idiopathic interstitial pneumonia         NaN  

[5 rows x 19717 columns]

### Multiple signatures of the same disease, average up and down genes

In [3]:
dis_dup ={}

for dis in dis_DEG.index:
    if dis not in dis_dup.keys():
        dis_dup[dis] =1 
    else:
        dis_dup[dis]+=1

print(len(dis_dup.keys())) #223
dis_dup

226


{'nicotine addiction': 1,
 'rhinovirus infection': 1,
 'dilated cardiomyopathy': 4,
 'bacterial infectious disease': 2,
 'idiopathic interstitial pneumonia': 2,
 'polycystic ovary syndrome': 12,
 'asthma': 18,
 'melanoma': 6,
 'bipolar disorder': 10,
 'childhood type dermatomyositis': 2,
 'rheumatoid arthritis': 4,
 'familial combined hyperlipidemia': 3,
 'nasopharynx carcinoma': 7,
 'uterine fibroid': 3,
 'cerebral palsy': 5,
 'bronchopulmonary dysplasia': 2,
 'urothelial carcinoma in situ': 1,
 'allergic contact dermatitis': 4,
 'septic shock': 1,
 'lung transplant rejection': 1,
 'cardiomyopathy': 4,
 'prostate cancer': 10,
 'dermatomyositis': 3,
 'dental cavity, complex': 1,
 'pre-eclampsia': 1,
 'actinic keratosis': 2,
 'acute t cell leukemia': 1,
 'schizophrenia': 19,
 'senescence': 2,
 'psoriasis vulgaris': 3,
 'nonspecific interstitial pneumonia': 1,
 'facioscapulohumeral muscular dystrophy': 4,
 'duchenne muscular dystrophy': 1,
 'pompe disease, infantile-onset form': 1,
 'pul

In [4]:
dis_DEG = dis_DEG.fillna(0)
print(dis_DEG.shape)
dis_DEG.head()

(555, 19717)


1110008P14Rik  1190002N15Rik  2'-PDE  \
nicotine addiction                           0.0            0.0     0.0   
rhinovirus infection                         0.0            0.0     0.0   
dilated cardiomyopathy                       0.0            0.0     0.0   
bacterial infectious disease                 0.0            0.0     0.0   
idiopathic interstitial pneumonia            0.0            0.0     0.0   

                                   2310036O22Rik  2900097C17Rik  7A5  A1BG  \
nicotine addiction                           0.0            0.0  0.0   0.0   
rhinovirus infection                         0.0            0.0  0.0   0.0   
dilated cardiomyopathy                       0.0            0.0  0.0   0.0   
bacterial infectious disease                 0.0            0.0  0.0   0.0   
idiopathic interstitial pneumonia            0.0            0.0  0.0   0.0   

                                   A1CF  A2BP1  A2M  ...  Zmiz2  Zranb2  \
nicotine addiction                  0.0    0.0  0.0  ...    0.0     0.0   
rhinovirus infection                0.0    0.0  0.0  ...    0.0     0.0   
dilated cardiomyopathy              0.0    0.0  0.0  ...    0.0     0.0   
bacterial infectious disease        0.0    0.0  0.0  ...    0.0     0.0   
idiopathic interstitial pneumonia   0.0    0.0  0.0  ...    0.0     0.0   

                                   Zwint  dJ222E13.2  hCG_15200  hCG_1806964  \
nicotine addiction                   0.0         0.0        0.0          0.0   
rhinovirus infection                 0.0         0.0        0.0          0.0   
dilated cardiomyopathy               0.0         0.0        0.0          0.0   
bacterial infectious disease         0.0         0.0        0.0          0.0   
idiopathic interstitial pneumonia    0.0         0.0        0.0          0.0   

                                   hCG_18290  tcag7.1196  tcag7.1228  \
nicotine addiction                       0.0         0.0         0.0   
rhinovirus infection                     0.0         0.0         0.0   
dilated cardiomyopathy                   0.0         0.0         0.0   
bacterial infectious disease             0.0         0.0         0.0   
idiopathic interstitial pneumonia        0.0         0.0         0.0   

                                   tcag7.1260  
nicotine addiction                        0.0  
rhinovirus infection                      0.0  
dilated cardiomyopathy                    0.0  
bacterial infectious disease              0.0  
idiopathic interstitial pneumonia         0.0  

[5 rows x 19717 columns]

In [5]:
ave_dis_DEG = pd.DataFrame(columns=dis_DEG.columns)

for dis in dis_dup.keys():
    tem_df = dis_DEG[dis_DEG.index==dis]
    print(dis)
    if dis_dup[dis] > 1:
        ave_dis_DEG=ave_dis_DEG.append(pd.Series(tem_df.mean(axis=0),name=dis),ignore_index=False)
        
    elif dis_dup[dis] ==1:
        ave_dis_DEG=ave_dis_DEG.append(tem_df,ignore_index=False)
    else:
        print("遍历疾病错误")
        
print(ave_dis_DEG.shape)
ave_dis_DEG.head()

nicotine addiction
rhinovirus infection
dilated cardiomyopathy
bacterial infectious disease
idiopathic interstitial pneumonia
polycystic ovary syndrome
asthma
melanoma
bipolar disorder
childhood type dermatomyositis
rheumatoid arthritis
familial combined hyperlipidemia
nasopharynx carcinoma
uterine fibroid
cerebral palsy
bronchopulmonary dysplasia
urothelial carcinoma in situ
allergic contact dermatitis
septic shock
lung transplant rejection
cardiomyopathy
prostate cancer
dermatomyositis
dental cavity, complex
pre-eclampsia
actinic keratosis
acute t cell leukemia
schizophrenia
senescence
psoriasis vulgaris
nonspecific interstitial pneumonia
facioscapulohumeral muscular dystrophy
duchenne muscular dystrophy
pompe disease, infantile-onset form
pulmonary hypertension
familial hypercholesterolemia
nephrosclerosis
williams-beuren syndrome
breast cancer
fragile x syndrome
nephroblastoma
testicular cancer
parasitic protozoa infectious disease
leukemia, chronic t-cell
gastrointestinal stromal 

1110008P14Rik  1190002N15Rik  2'-PDE  \
nicotine addiction                           0.0            0.0     0.0   
rhinovirus infection                         0.0            0.0     0.0   
dilated cardiomyopathy                       0.0            0.0     0.0   
bacterial infectious disease                 0.0            0.0     0.0   
idiopathic interstitial pneumonia            0.0            0.0     0.0   

                                   2310036O22Rik  2900097C17Rik  7A5  A1BG  \
nicotine addiction                           0.0            0.0  0.0   0.0   
rhinovirus infection                         0.0            0.0  0.0   0.0   
dilated cardiomyopathy                       0.0            0.0  0.0   0.0   
bacterial infectious disease                 0.0            0.0  0.0   0.0   
idiopathic interstitial pneumonia            0.0            0.0  0.0   0.0   

                                   A1CF  A2BP1       A2M  ...  Zmiz2  Zranb2  \
nicotine addiction                  0.0    0.0  0.000000  ...    0.0     0.0   
rhinovirus infection                0.0    0.0  0.000000  ...    0.0     0.0   
dilated cardiomyopathy              0.0    0.0 -0.031713  ...    0.0     0.0   
bacterial infectious disease        0.0    0.0  0.007743  ...    0.0     0.0   
idiopathic interstitial pneumonia   0.0    0.0  0.034011  ...    0.0     0.0   

                                   Zwint  dJ222E13.2  hCG_15200  hCG_1806964  \
nicotine addiction                   0.0         0.0        0.0          0.0   
rhinovirus infection                 0.0         0.0        0.0          0.0   
dilated cardiomyopathy               0.0         0.0        0.0          0.0   
bacterial infectious disease         0.0         0.0        0.0          0.0   
idiopathic interstitial pneumonia    0.0         0.0        0.0          0.0   

                                   hCG_18290  tcag7.1196  tcag7.1228  \
nicotine addiction                       0.0         0.0         0.0   
rhinovirus infection                     0.0         0.0         0.0   
dilated cardiomyopathy                   0.0         0.0         0.0   
bacterial infectious disease             0.0         0.0         0.0   
idiopathic interstitial pneumonia        0.0         0.0         0.0   

                                   tcag7.1260  
nicotine addiction                        0.0  
rhinovirus infection                      0.0  
dilated cardiomyopathy                    0.0  
bacterial infectious disease              0.0  
idiopathic interstitial pneumonia         0.0  

[5 rows x 19717 columns]

In [6]:
# ave_dis_DEG.to_csv("ave_dis_DEG_name.csv")

In [39]:
"""Number of diseases after averaging：
    name index：226
    CUI index：223"""
# print(len(ave_dis_DEG))

223


### Count the occurrences of each gene in all diseases
#### <font color=red>Take the number of occurrences as a threshold</font>，Screening genes<br>
### The size of the feature matrix----name as index：（226，19717），CUI as index：（223，19563）

In [40]:
genes_dup ={}

for gene in dis_DEG.columns:
    genes_dup[gene] = 0
    for occ in list(ave_dis_DEG[gene]):
        if occ != 0:
            genes_dup[gene] +=1

genes_dup

{'1110008P14Rik': 1,
 '1190002N15Rik': 1,
 "2'-PDE": 1,
 '2310036O22Rik': 1,
 '2900097C17Rik': 1,
 '7A5': 1,
 'A1BG': 2,
 'A1CF': 5,
 'A2BP1': 1,
 'A2M': 59,
 'A2M-AS1': 10,
 'A2ML1': 3,
 'A4GALT': 1,
 'AACS': 17,
 'AADAC': 17,
 'AADACL2': 5,
 'AADAT': 1,
 'AAED1': 5,
 'AAGAB': 1,
 'AAK1': 1,
 'AAMDC': 1,
 'AAMP': 6,
 'AANAT': 1,
 'AAR2': 1,
 'AARD': 3,
 'AARS': 44,
 'AASDH': 2,
 'AASDHPPT': 4,
 'AASS': 6,
 'AATF': 5,
 'AATK': 14,
 'ABAT': 14,
 'ABBA-1': 1,
 'ABCA1': 11,
 'ABCA10': 1,
 'ABCA11P': 1,
 'ABCA12': 10,
 'ABCA13': 10,
 'ABCA17P': 4,
 'ABCA2': 2,
 'ABCA3': 8,
 'ABCA5': 8,
 'ABCA6': 9,
 'ABCA7': 2,
 'ABCA8': 25,
 'ABCA9': 6,
 'ABCB1': 11,
 'ABCB10': 6,
 'ABCB11': 3,
 'ABCB4': 20,
 'ABCB6': 1,
 'ABCB7': 5,
 'ABCB8': 3,
 'ABCB9': 2,
 'ABCC1': 7,
 'ABCC10': 4,
 'ABCC11': 2,
 'ABCC12': 2,
 'ABCC13': 4,
 'ABCC2': 6,
 'ABCC3': 8,
 'ABCC4': 7,
 'ABCC5': 9,
 'ABCC8': 2,
 'ABCC9': 8,
 'ABCD1': 1,
 'ABCD2': 4,
 'ABCD3': 18,
 'ABCD4': 2,
 'ABCE1': 10,
 'ABCF1': 13,
 'ABCF2': 2,
 'ABCG1':

In [41]:
print(len(genes_dup))

19563


In [43]:
count = 0
count_threshold = 0
_max= 0 
fea_genes=[]

for gene in genes_dup:
    if genes_dup[gene]==0:
        count+=1
    if genes_dup[gene]>=(100):
        count_threshold+=1
        fea_genes.append(gene)
    if genes_dup[gene]>_max:
        _max = genes_dup[gene]
        
print("supposed:0 ---",count)
print(count_threshold)
print(_max)

supposed:0 --- 0
267
175


#### 1.Test 100 as the threshold,nums of disease:
    1.id 267
    2.name 284

In [44]:
print(len(fea_genes))

267


In [45]:
fea_dis = ave_dis_DEG[fea_genes]
fea_dis.head()

ACTB     ACTG1     ALDH2     ALDOA    ANP32B     ANXA1    ANXA2  \
C0028043  0.064248  0.057393  0.000000 -0.013307 -0.008606  0.000000  0.00000   
C0276447 -0.018748 -0.066498  0.017732  0.014883  0.000000 -0.024498  0.00000   
C0007193 -0.005538 -0.003351 -0.009115  0.014854  0.000000  0.000000  0.00000   
C0004623  0.073810  0.040390 -0.001929  0.009611  0.005133  0.000000  0.01442   
C0085786 -0.063654  0.000000 -0.019904  0.000000  0.000000 -0.015101  0.00000   

             ANXA5   ARL6IP5     ARPC2  ...     UBA52       UBB       UBC  \
C0028043  0.000000  0.000000 -0.014727  ...  0.076123 -0.015261  0.066425   
C0276447 -0.049095  0.021049  0.014442  ...  0.013136  0.000000  0.000000   
C0007193  0.010102 -0.015487 -0.003529  ...  0.003684  0.028295  0.043564   
C0004623  0.015189  0.000000  0.018244  ...  0.007298  0.000000  0.008932   
C0085786  0.000000  0.000000  0.005371  ...  0.011232  0.000000 -0.034063   

             UQCRH     UQCRQ       VIM      YBX1     YWHAQ     YWHAZ     ZFP36  
C0028043  0.077111  0.012225  0.000000  0.104099 -0.022610  0.018012  0.000000  
C0276447 -0.017052  0.000000  0.000000  0.000000 -0.015306  0.000000  0.015338  
C0007193 -0.009113  0.000000  0.007928  0.000000  0.000000  0.000000 -0.002928  
C0004623  0.007242  0.000000  0.058038 -0.011562  0.010378  0.008126 -0.011766  
C0085786 -0.025133 -0.012593  0.000000 -0.022423  0.007313  0.021626 -0.112035  

[5 rows x 267 columns]

In [46]:
#fea_dis.to_csv('fea_dis_name.csv',index=True)
#fea_dis.to_csv('fea_dis_id.csv',index=True)

In [47]:
max_up = fea_dis.max(axis=0)
max_down = fea_dis.min(axis=0)
print(max_up.max())
print(max_down.min())

0.6450769305229187
-0.4551387131214142


### Select diseases with similarity labels
    1.UMLS-CUI id ：138
    2.CREEDS_dis_name ：136

In [3]:
dis = pd.read_csv("用作数据的疾病目录_id.csv")
print(len(dis))
dis.head()

138


CREEDS_dis_name  UMLS-CUI
0        dilated cardiomyopathy  C0007193
1  bacterial infectious disease  C0004623
2     polycystic ovary syndrome  C0032460
3                        asthma  C0004096
4                      melanoma  C0025202

In [51]:
fea_withlabel = pd.DataFrame(columns=fea_dis.columns)
for dis in dis["UMLS-CUI"]:
    fea_tem = fea_dis[fea_dis.index==dis]
    fea_withlabel = fea_withlabel.append(fea_tem,ignore_index=False)
print(len(fea_withlabel)) #138
fea_withlabel.head()

138


ACTB     ACTG1     ALDH2     ALDOA    ANP32B    ANXA1     ANXA2  \
C0007193 -0.005538 -0.003351 -0.009115  0.014854  0.000000  0.00000  0.000000   
C0004623  0.073810  0.040390 -0.001929  0.009611  0.005133  0.00000  0.014420   
C0032460  0.005896 -0.019739  0.000000  0.005699 -0.000403  0.00800  0.001595   
C0004096 -0.011333 -0.035207 -0.006853  0.001647 -0.002864 -0.00133  0.005973   
C0025202  0.022291 -0.001581 -0.034077  0.013051  0.000000 -0.00664 -0.014729   

             ANXA5   ARL6IP5     ARPC2  ...     UBA52       UBB       UBC  \
C0007193  0.010102 -0.015487 -0.003529  ...  0.003684  0.028295  0.043564   
C0004623  0.015189  0.000000  0.018244  ...  0.007298  0.000000  0.008932   
C0032460  0.003025  0.000647  0.004306  ... -0.011284 -0.000242  0.018490   
C0004096  0.002035  0.000858 -0.005306  ...  0.004623 -0.003044 -0.012178   
C0025202  0.009445  0.003740  0.004085  ... -0.003697  0.000000 -0.005839   

             UQCRH     UQCRQ       VIM      YBX1     YWHAQ     YWHAZ     ZFP36  
C0007193 -0.009113  0.000000  0.007928  0.000000  0.000000  0.000000 -0.002928  
C0004623  0.007242  0.000000  0.058038 -0.011562  0.010378  0.008126 -0.011766  
C0032460 -0.008182  0.002951 -0.001736 -0.003992 -0.002798 -0.002718  0.001739  
C0004096 -0.001064  0.007969 -0.004879 -0.009546 -0.006188 -0.002771 -0.020539  
C0025202  0.002282  0.000000  0.003155  0.015831  0.000000  0.004267 -0.010503  

[5 rows x 267 columns]

In [52]:
#fea_withlabel.to_csv("fea_withlabel_id.csv")